tests/saml/test_parser.py

```diff
@@ -0,0 +1,287 @@
+from nose.tools import raises, eq_
+
+from api.saml.metadata import IdentityProviderMetadata, UIInfo, LocalizableMetadataItem, Service, \
+    ServiceProviderMetadata, NameIDFormat
+from api.saml.parser import SAMLMetadataParsingError, SAMLMetadataParser
+from tests.saml import fixtures
+
+
+class SAMLMetadataParserTest(object):
+    def _strip_certificate(self, certificate):
+        """
+        Converts certificate to a one-line format
+
+        :param certificate: Certificate in a multi-line format
+        :type certificate: string
+
+        :return: Certificate in a one-line format
+        :rtype: string
+        """
+
+        return certificate.replace('\n', '')
+
+    def _check_idp_metadata(
+            self,
+            idp_metadata,
+            entity_id,
+            ui_info,
+            name_id_format,
+            sso_service,
+            want_authn_requests_signed,
+            signing_certificates,
+            encryption_certificates):
+        assert isinstance(idp_metadata, IdentityProviderMetadata)
+        assert idp_metadata.entity_id == entity_id
+
+        assert isinstance(idp_metadata.ui_info, UIInfo)
+        assert idp_metadata.ui_info == ui_info
+
+        assert idp_metadata.name_id_format == name_id_format
+
+        assert idp_metadata.sso_service == sso_service
+
+        assert idp_metadata.want_authn_requests_signed == want_authn_requests_signed
+
+        assert idp_metadata.signing_certificates == signing_certificates
+        assert idp_metadata.encryption_certificates == encryption_certificates
+
+    @raises(SAMLMetadataParsingError)
+    def test_parse_raises_exception_when_xml_metadata_has_incorrect_format(self):
+        # Arrange
+        metadata_parser = SAMLMetadataParser()
+
+        # Act
+        metadata_parser.parse(fixtures.INCORRECT_XML)
+
+    @raises(SAMLMetadataParsingError)
+    def test_parse_raises_exception_when_idp_metadata_does_not_contain_sso_service(self):
+        # Arrange
+        metadata_parser = SAMLMetadataParser()
+
+        # Act
+        metadata_parser.parse(fixtures.INCORRECT_ONE_IDP_METADATA_WITHOUT_SSO_SERVICE)
+
+    @raises(SAMLMetadataParsingError)
+    def test_parse_raises_exception_when_idp_metadata_contains_sso_service_with_wrong_binding(self):
+        # Arrange
+        metadata_parser = SAMLMetadataParser()
+
+        # Act
+        metadata_parser.parse(fixtures.INCORRECT_ONE_IDP_METADATA_WITH_SSO_SERVICE_WITH_WRONG_BINDING)
+
+    def test_parse_correctly_parses_one_idp_metadata(self):
+        # Arrange
+        metadata_parser = SAMLMetadataParser()
+
+        # Act
+        result = metadata_parser.parse(fixtures.CORRECT_ONE_IDP_METADATA)
+
+        # Assert
+        assert isinstance(result, list)
+        eq_(len(result), 1)
+
+        eq_(
+            result[0],
```

Comment:
It's not necessary to change this, but another cosmetic note: in cases like this I like to assign the list to a variable.

```
[result] = result
eq_(result, ...)
```

This proves that `result` is a list containing a single item without needing an extra explicit assertion.

Reply:
Sure, changed it in accordance with your comment